finance/table_finance_us_sec_public_company.go

```diff
@@ -19,18 +21,87 @@ func tableFinanceUsSecPublicCompany(ctx context.Context) *plugin.Table {
+			{Name: "currency", Type: proto.ColumnType_STRING, Description: "Currency the symbol is traded in using."},
+			{Name: "is_enabled", Type: proto.ColumnType_BOOL, Description: "True if the symbol is enabled for trading on IEX."},
+			{Name: "date", Type: proto.ColumnType_TIMESTAMP, Description: "Date the symbol reference data was generated."},
+			{Name: "exchange", Type: proto.ColumnType_STRING, Description: "Exchange symbol."},
+			{Name: "exchange_name", Type: proto.ColumnType_STRING, Description: "Exchange name."},
+			{Name: "exchange_segment", Type: proto.ColumnType_STRING, Description: "Exchange segment."},
+			{Name: "exchange_segment_name", Type: proto.ColumnType_STRING, Description: "Exchange segment name."},
+			{Name: "exchange_suffix", Type: proto.ColumnType_STRING, Description: "Exchange segment suffix."},
+			{Name: "figi", Type: proto.ColumnType_STRING, Transform: transform.FromField("FIGI"), Description: "OpenFIGI id for the security, if available."},
+			{Name: "iex_id", Type: proto.ColumnType_STRING, Transform: transform.FromField("IEXID"), Description: "Unique ID applied by IEX to track securities through symbol changes."},
+			{Name: "lei", Type: proto.ColumnType_STRING, Transform: transform.FromField("LEI"), Description: "Legal Entity Identifier (LEI) for the security, if available."},
+			{Name: "region", Type: proto.ColumnType_STRING, Description: "Country code for the symbol."},
+			{Name: "type", Type: proto.ColumnType_STRING, Description: "common issue typepossible values are: ad - ADR, cs - Common Stock, cef - Closed End Fund, et - ETF, oef - Open Ended Fund, ps - Preferred Stock, rt - Right, struct - Structured Product, ut - Unit, wi - When Issued, wt - Warrant, empty - Other."},
 		},
 	}
 }
 
+type Company struct {
+	CIK                 string `json:"cik"`
+	Currency            string `json:"currency"`
+	Date                string `json:"date"`
+	Exchange            string `json:"exchange"`
+	ExchangeName        string `json:"exchangeName"`
+	ExchangeSegment     string `json:"exchangeSegment"`
+	ExchangeSegmentName string `json:"exchangeSegmentName"`
+	ExchangeSuffix      string `json:"exchangeSuffix"`
+	FIGI                string `json:"figi"`
+	IEXID               string `json:"iexId"`
+	IsEnabled           bool   `json:"isEnabled"`
+	LEI                 string `json:"lei"`
+	Name                string `json:"name"`
+	Region              string `json:"region"`
+	Symbol              string `json:"symbol"`
+	Type                string `json:"type"`
+	// Define other fields based on the API response
+}
+
 func listUsSecPublicCompany(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
-	companies, err := core.GetPublicCompaniesWithHeaders(map[string]string{"User-Agent": "Steampipe/v0.x"})
+
+	client := resty.New()
```

Comment:
Please try using a popular SDK if one exists

Reply:
Updated the package to make the API call.